rfcs/repo-management/shield-issue-expectations.md

```diff
@@ -0,0 +1,99 @@
+# RFC: Expectations on issue resolution while on Shield
+
+---
+
+Contributors: @khmakoto, @JustSlone, @gouttierre
+
+## Summary
+
+This RFC seeks to improve the Shield process by defining better expectations regarding our response to issues and outlining key actions to follow as part of the rotation.
+
+## Problem statement
+
+Since convergence work started our Shield process has been continuously changing to adapt to the changing conditions in our team (merging of the libraries in one repo, influx and efflux of people in the team, balancing v0, v8 and converged work, etc). This has generated enough instability that, coupled with an already vague set of expectations around issue resolution, has amounted to a somewhat unmanageable situation where the issue number keeps creeping up, the person on point for Shield that month feels like they are not stepping up to the task, and the people logging the issues feeling frustrated about the delay in response/their issues not being given a solution.
```

Comment:
After only 3 weeks on Shield, I feel this too.  I think there is a hidden cost of context switching between monitoring the teams channel & communicating back on issues, reviewing PRs from outside the immediate team, and working on the backlog of bugs.  This leads to feeling like there is an ephemeral SLA of instant response that isn't being met.

Proposed solution:

- Have 3 people "on shield" at a time.  One primary and two secondary shields.  
- The primary shields the team from bug debt.  
  - They focus solely on reproducing, investigating, and closing bugs. 
  - The SLA on investigating a bug is 1 week.
  - The SLA on resolving a bug (either fixed, low-priority someday, or won't fix) is 1 month/current sprint).
- One secondary is on communications.  
  - They monitor the team channel and investigate/respond.  
  - They try to encourage the community to do their own PRs (as mentioned in this RFC).  
   - They may go ping teammates if someone on the teams channel asks for an update on an issue or PR.  
   - The SLA is to respond to a teams post is 1 business day
- The other secondary is on PRs.  
  - They don't have to do all the the reviews, but they make sure that community PRs are being addressed in a timely manner by the code owner on the team.
   - The SLA is 1 week for getting some feedback on the PR.
   - The SLA for merge after all issues resolved is 1 week.
   
Benefits 
- The primary doesn't have to context switch
- The primary has a clear metric to reduce bug debt.  We can measure reduced bugs vs new bugs coming in.
- The secondary duties shouldn't be taking more than an hour or two per day at most.